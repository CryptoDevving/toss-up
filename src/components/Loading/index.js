import dice_logo from "../../assets/dice_logo.png";
import dice_rolling from "../../assets/eth.gif";
import React, { Component } from "react";
import eth from "../../assets/eth.png";
import FadeIn from "react-fade-in";
import "../App.css";

class Loading extends Component {
  render() {
    return (
      <FadeIn>
        <div
          className="container-fluid mt-5 col-m-4"
          style={{ maxWidth: "550px" }}
        >
          <div className="col-sm">
            <main
              role="main"
              className="col-lg-12 text-monospace text-center text-light"
            >
              <div className="content mr-auto ml-auto">
                <div id="content" className="mt-3">
                  <div className="card card-2 mb-4">
                    <div className="card-body">
                      <div>
                        <img src={dice_rolling} width="225" alt="logo" />
                      </div>
                      &nbsp;
                      <p></p>
                      <div className="input-group mb-4">
                        <input
                          id="disabledInput"
                          type="text"
                          className="form-control form-control-md"
                          placeholder="rolling..."
                          disabled
                        />
                        <div className="input-group-append">
                          <div className="input-group-text">
                            <img src={eth} height="20" alt="" />
                            &nbsp;<b>ETH</b>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div className="numbers">
                      {!this.props.balance ? (
                        <div
                          id="loader"
                          className="spinner-border float-right"
                          role="status"
                        ></div>
                      ) : (
                        <div
                          className="float-right"
                          style={{ width: "220px", padding: "7px" }}
                        >
                          <div
                            className="float-left"
                            style={{ height: "17px" }}
                          >
                            <b>MaxBet: </b>
                          </div>
                          <div
                            className="float-right"
                            style={{ height: "17px" }}
                          >
                            {Number(
                              this.props.web3.utils.fromWei(
                                this.props.maxBet.toString()
                              )
                            ).toFixed(5)}{" "}
                            <b>ETH </b>
                          </div>
                          <br></br>
                          <div
                            className="float-left"
                            style={{ height: "17px" }}
                          >
                            <b>MinBet: </b>
                          </div>
                          <div
                            className="float-right"
                            style={{ height: "17px" }}
                          >
                            {Number(
                              this.props.web3.utils.fromWei(
                                this.props.minBet.toString()
                              )
                            ).toFixed(5)}{" "}
                            <b>ETH </b>
                          </div>
                          <br></br>
                          <div className="float-left">
                            <b>Balance: </b>
                          </div>
                          <div className="float-right">
                            {Number(
                              this.props.web3.utils.fromWei(
                                this.props.balance.toString()
                              )
                            ).toFixed(5)}{" "}
                            <b>ETH </b>
                          </div>
                        </div>
                      )}
                      <div className="dice2">
                        <img src={dice_logo} width="80" alt="logo" />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </main>
          </div>
        </div>
      </FadeIn>
    );
  }
}

export default Loading;
