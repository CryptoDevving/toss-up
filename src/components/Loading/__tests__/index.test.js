import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';

import Loading from '..';

afterEach(cleanup);

describe('Loading component', () => {
  it('renders', () => {
    render(<Loading />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Loading />);

    expect(asFragment()).toMatchSnapshot();
  });

})