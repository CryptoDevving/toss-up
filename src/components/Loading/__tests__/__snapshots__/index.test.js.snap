// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
    >
      <div
        class="container-fluid mt-5 col-m-4"
        style="max-width: 550px;"
      >
        <div
          class="col-sm"
        >
          <main
            class="col-lg-12 text-monospace text-center text-light"
            role="main"
          >
            <div
              class="content mr-auto ml-auto"
            >
              <div
                class="mt-3"
                id="content"
              >
                <div
                  class="card card-2 mb-4"
                >
                  <div
                    class="card-body"
                  >
                    <div>
                      <img
                        alt="logo"
                        src="eth.gif"
                        width="225"
                      />
                    </div>
                     
                    <p />
                    <div
                      class="input-group mb-4"
                    >
                      <input
                        class="form-control form-control-md"
                        disabled=""
                        id="disabledInput"
                        placeholder="rolling..."
                        type="text"
                      />
                      <div
                        class="input-group-append"
                      >
                        <div
                          class="input-group-text"
                        >
                          <img
                            alt=""
                            height="20"
                            src="eth.png"
                          />
                           
                          <b>
                            ETH
                          </b>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="numbers"
                  >
                    <div
                      class="spinner-border float-right"
                      id="loader"
                      role="status"
                    />
                    <div
                      class="dice2"
                    >
                      <img
                        alt="logo"
                        src="dice_logo.png"
                        width="80"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </main>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
